File: Patterns/Java/Creational/Builder/BuilderVestBeret.java
public class BuilderVestBeret extends Builder{

    BuilderVestBeret(){
        outfit = new Outfit();
    }

    public void buildCoat(){
        outfit.addClothe( new Vest() );
    }


    public void buildHat(){
        outfit.addClothe( new Beret() );
    }

    public Outfit getOutfit(){
        return outfit;
    }

}

File: Patterns/C#/Creational/Builder/BuilderVestBeret.cs
public class BuilderVestBeret : Builder{

    public BuilderVestBeret(){
        outfit = new Outfit();
    }

    public override void buildCoat(){
        outfit.addClothe( new Vest() );
    }

    public override void buildHat(){
        outfit.addClothe( new Beret() );
    }

    public override Outfit getOutfit(){
        return outfit;
    }

}
